internal/pgadmin/users.go

```diff
@@ -72,11 +74,47 @@ if sys.path[0] != root:
 	//
 	// TODO(cbandy): pgAdmin v4.21 adds "auth_source" and "username" as required attributes.
 	// - https://git.postgresql.org/gitweb/?p=pgadmin4.git;f=web/pgadmin/model/__init__.py;hb=REL-4_21#l65
+
+	// When the users are created or modified, server groups and connections will
+	// also be configured, similar to the way server groups and connections are
+	// made with their respective dialog windows.
+	// - https://www.pgadmin.org/docs/pgadmin4/latest/server_group_dialog.html
+	// - https://www.pgadmin.org/docs/pgadmin4/latest/server_dialog.html
+
+	// We use a similar method to the import method when creating server connections
+	// - https://www.pgadmin.org/docs/pgadmin4/development/import_export_servers.html
+	// - https://git.postgresql.org/gitweb/?p=pgadmin4.git;a=blob;f=web/setup.py;h=eb7d9044521a65781e6bdfebf02c8b2572f1dc97;hb=REL-4_20#l256
+
+	// One server connection will be configured for each defined user.
+	// The name of the server connection will be the same as the cluster name.
+	// Note that the server connections are created when the users are created or
+	// modified. Changes to a server connection will generally persist until a
+	// change is made to the corresponding user. For custom server connections,
+	// a new server should be created with a unique name.
+
+	// The server connection password is the plaintext password encrypted with the
+	// password itself as the key.
+	// - https://git.postgresql.org/gitweb/?p=pgadmin4.git;a=blob;f=web/pgadmin/__init__.py;h=820c8015a1ca69703d7089783bc10ebcdb307f1c;hb=REL-4_20#l580
+	// - https://git.postgresql.org/gitweb/?p=pgadmin4.git;a=blob;f=web/pgadmin/utils/master_password.py;h=759bf36e0615b479286d04e29a2aa057689eab17;hb=REL-4_20#l20
+	// - https://git.postgresql.org/gitweb/?p=pgadmin4.git;a=blob;f=web/pgadmin/browser/server_groups/servers/__init__.py;h=3c96c1051342840cb6ec1daf15d0559cb23f836c;hb=REL-4_20#l840
+
+	// Due to limitations on the types of updates that can be made to active server
+	// connections, when the current server connection is updated, we need to delete
+	// it and add a new server connection in its place. This will require a refresh
+	// if pgAdmin web GUI is being used when the update takes place.
```

Comment:
📝 I think I want more here than just "limitations", but I'm not sure what exactly. If I figure it out, I'll open my own PR. 😄 

💯 👍🏻 The link perfectly summarizes the issue.